clang/lib/Sema/SemaOverload.cpp

```diff
@@ -10085,10 +10085,14 @@ getImplicitObjectParamType(ASTContext &Context, const FunctionDecl *F) {
   return M->getFunctionObjectParameterReferenceType();
 }
 
-static bool haveSameParameterTypes(ASTContext &Context, const FunctionDecl *F1,
+static bool allowAmbiguityWithSelf(ASTContext &Context, const FunctionDecl *F1,
```

Comment:
Could you elaborate why `declaresSameEntity` is not enough and we need extra checks for parameter types?
I am asking because if it's not enough for regular functions, maybe there is a similar case for function templates.
And the name is a bit misleading as `F1` and `F2` can clearly be two different functions, so it is unclear what `self`  refers to.

Reply:
I added a todo to get rid of the parameter type checks. I do not see any test failures without it.
Since this could again be a potentially breaking change, I will try cleaning this up in a separate PR.

Changed the name to `allowAmbiguity`.

Reply:
`allowAmbiguity` looks fine, thanks!